### PR TITLE
GitHub Actions: Remove actions-rs/toolchain which is end-of-life

### LINKED ISSUE #1873: GitHub Actions: actions-rs/toolchain is end-of-life and should be replaced
https://github.com/casey/just/blob/ebead95ac990db1257c7a46198e43c5bb0def593/.github/workflows/ci.yaml#L26
* https://github.com/actions-rs/toolchain is archived.
* https://docs.github.com/en/actions/automating-builds-and-tests currently has no Rust solution.
* https://github.com/actions-rust-lang/setup-rust-toolchain seems to be a full-featured solution.
* GitHub Actions base images have rustup v1.26.0,  cargo v1.75.0, rustc v1.75.0, and rustfmt 1.7.0 pre-installed.

### PR DESCRIPTION
**Pull request by cclauss**
Fixes: #1873 
* #1873 

Use the version of `cargo` that is pre-installed in the GHA base image.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -49,13 +49,6 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
-    - name: Install Rust Toolchain Components
-      uses: actions-rs/toolchain@v1
-      with:
-        profile: minimal
-        target: ${{ matrix.target }}
```
Comment (cclauss) [inline .github/workflows/release.yaml:56]: This line bugs me...  How can we communicate the profile and target to the Rust Toolchain?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -118,13 +106,6 @@ jobs:
         icacls C:\Windows\System32\bash.exe /grant administrators:F
         del C:\Windows\System32\bash.exe
 
-    - name: Install Rust Toolchain Components
-      uses: actions-rs/toolchain@v1
-      with:
-        components: clippy, rustfmt
-        override: true
```
Comment (cclauss) [inline .github/workflows/ci.yaml:125]: How can we communicate override to the Rust Toolchain?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (casey) [reply .github/workflows/release.yaml:56]: Is `rustup` installed on the base image? If so, can we use `rustup` to install the correct toolchain?

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (casey) [reply .github/workflows/ci.yaml:125]: The `override` part? I'm not 100% sure it's being used, so we could just see if it works without it.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (cclauss) [reply .github/workflows/release.yaml:56]: Yes, it is.  See the last line of #1873.  I know little about Rust tooling so you probably need to take over this pull request to get the `release` job finished.

It is cool to see that the `ci` job just works.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (cclauss) [reply .github/workflows/ci.yaml:125]: The `ci` job is green ✅

**Comment by casey** [comment]
Installing the right toolchain in `bin/package` turned out to be super easy. Great PR! Removing a dependency is always awesome, and this should make our CI job faster.

**Pull request merged**